### PR TITLE
Translate some function-like C macros

### PR DESCRIPTION
**Pull request by JoshuaLampert**
Explicitly tell Clang.jl to translate some useful function-like macros.

### CODE REVIEW

File: src/LibP4est_v2.jl [hunk 1]
```diff
@@ -7080,9 +7080,9 @@ function Base.getproperty(x::Ptr{p6est_quadrant_data}, f::Symbol)
     f === :user_long && return Ptr{Clong}(x + 0)
     f === :user_int && return Ptr{Cint}(x + 0)
     f === :which_tree && return Ptr{p4est_topidx_t}(x + 0)
-    f === :piggy1 && return Ptr{__JL_Ctag_317}(x + 0)
-    f === :piggy2 && return Ptr{__JL_Ctag_318}(x + 0)
-    f === :piggy3 && return Ptr{__JL_Ctag_319}(x + 0)
+    f === :piggy1 && return Ptr{__JL_Ctag_320}(x + 0)
+    f === :piggy2 && return Ptr{__JL_Ctag_315}(x + 0)
+    f === :piggy3 && return Ptr{__JL_Ctag_316}(x + 0)
```
Comment (ranocha) [inline src/LibP4est_v2.jl:7085]: Are these `__JL_Ctag_*` with different numbers basically the same structs?

File: src/LibP4est_v2.jl [hunk 2]
```diff
@@ -13063,6 +13063,8 @@ const P4EST_QCOORD_MAX = INT32_MAX
 
 const P4EST_QCOORD_1 = p4est_qcoord_t(1)
 
+P4EST_QCOORD_ABS(x) = (p4est_qcoord_t(labs))(Clong(x))
```
Comment (ranocha) [inline src/LibP4est_v2.jl:13066]: What is `labs` (here and below) - and `llabs` below?

File: src/LibP4est_v2.jl (hunk 1 above)
Comment (JoshuaLampert) [reply src/LibP4est_v2.jl:7085]: Yes, they are the same. I don't know why we need those twice each.

File: src/LibP4est_v2.jl (hunk 2 above)
Comment (JoshuaLampert) [reply src/LibP4est_v2.jl:13066]: You are right. I remove these functions again. We probably don't need them.

File: src/LibP4est_v2.jl (hunk 1 above)
Comment (ranocha) [reply src/LibP4est_v2.jl:7085]: Thanks!

**Review by ranocha — APPROVED** [review]
Thanks! This looks good to me.

When you think this is ready to be merged, please mark this PR as ready for review and ping me so that I can trigger the merge.

**Comment by ranocha** [comment]
In the bottom, you can see that we cannot merge this PR right now since it is still marked as draft. You need to change that to allow merging it.

image

**Comment by JoshuaLampert** [comment]
Done @ranocha

**Comment by ranocha** [comment]
Thanks!

**Pull request merged**